### PR TITLE
chore(typescript): convert cli.js to cli.ts

### PR DESCRIPTION
**Pull request by sjelin**

### CODE REVIEW

File: lib/cli.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    .alias('name', 'capabilities.name')
+    .alias('platform', 'capabilities.platform')
+    .alias('platform-version', 'capabilities.version')
+    .alias('tags', 'capabilities.tags')
+    .alias('build', 'capabilities.build')
+    .alias('grep', 'jasmineNodeOpts.grep')
+    .alias('invert-grep', 'jasmineNodeOpts.invertGrep')
+    .alias('explorer', 'elementExplorer')
+    .string('capabilities.tunnel-identifier')
+    .check(function(arg: ProtractorArgs) {
```
Comment (juliemr) [inline lib/cli.ts]: I don't think this type is right - this is the leftover args, which is just an array of strings.

File: lib/cli.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      if (arg._.length > 1) {
+        throw 'Error: more than one config file specified';
+      }
+    });
+
+var argv: ProtractorArgs = optimist.parse(args);
```
Comment (juliemr) [inline lib/cli.ts]: I'm not sure typing helps here, since this is an arbitrary object (since the user could include any flags)

File: lib/cli.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+if (!configFile && !argv.elementExplorer && args.length < 3) {
+  console.log(
+      '**you must either specify a configuration file ' +
+      'or at least 3 options. See below for the options:\n');
+  optimist.showHelp();
+  process.exit(1);
+}
+
+// Run the launcher
+require('./launcher').init(configFile, argv);
```
Comment (juliemr) [inline lib/cli.ts]: Change to:
```js
import {init} from './launcher';
init(configFile, argv);
```

File: lib/cli.ts (hunk 2 above)
Comment (sjelin) [inline lib/cli.ts]: Would you prefer `any` then?

File: lib/cli.ts (hunk 1 above)
Comment (sjelin) [inline lib/cli.ts]: It definitely isn't an array of strings.  I didn't change the `if` statement on the line below.  Maybe it's just `{_: string[]}` though

File: lib/cli.ts (hunk 2 above)
Comment (juliemr) [inline lib/cli.ts]: Yup.

File: lib/cli.ts (hunk 1 above)
Comment (juliemr) [inline lib/cli.ts]: My bad, I missed that. Yes, you're right. But if we're removing the type below, doesn't really matter.

File: lib/cli.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+if (!configFile && !argv.elementExplorer && args.length < 3) {
+  console.log(
+      '**you must either specify a configuration file ' +
+      'or at least 3 options. See below for the options:\n');
+  optimist.showHelp();
+  process.exit(1);
+}
+
+// Run the launcher
+import * as launcher from './launcher';
```
Comment (juliemr) [inline lib/cli.ts]: Maybe we should move the import to the top of the file.

**Pull request merged**